renderer-android-view/public/src/androidMain/kotlin/software/amazon/app/platform/renderer/ViewRenderer.kt

```diff
@@ -145,9 +147,12 @@ public abstract class ViewRenderer<in ModelT : BaseModel> : BaseAndroidViewRende
     lastModel = null
 
     // Remove the view from the parent. In case the Renderer is reused we inflate a new
-    // View and add it to the parent.
-    if (view.parent === parent) {
-      parent.removeView(view)
+    // View and add it to the parent. Do this on the main thread to avoid colliding with Android's
+    // view cleanup.
+    CoroutineScope(Dispatchers.Main).launch {
+      if (view.parent === parent) {
+        parent.removeView(view)
+      }
```

Comment:
A test is missing. You were finally able to reliably reproduce the crash we've seen in the field and this can be covered with a test! I created one here: https://github.com/amzn/app-platform/pull/151 It fails today, but with your fix it is green.